src/main/kotlin/JoinDialog.kt

```diff
@@ -5,25 +5,34 @@ import tornadofx.*
 
 data class JoinDetails(val channel: String)
 
-class JoinDetailsModel : ItemViewModel<JoinDetails>() {
-    private val controller: MainController by inject()
+class JoinDialogController(private val controller: MainController) : Component() {
```

Comment:
Might need to have an interface for this so we can pass a mocked one into JoinDetailsModel for testing